Trimble.Modus.Components/Controls/TMSwitch/TMSwitch.xaml.cs

```diff
@@ -0,0 +1,210 @@
+            }
+        }
+    }
+    private void UpdateSwitch(TMSwitch tMSwitch)
+    {
+
+        tapGestureRecognizer = new TapGestureRecognizer();
+        tapGestureRecognizer.Tapped += OnSwitchTapped;
+        border.GestureRecognizers.Add(tapGestureRecognizer);
+        tMSwitch.border.BackgroundColor = ResourcesDictionary.ColorsDictionary(ColorsConstants.SwitchUnselected);
+        tMSwitch.border.Padding = 5;
+    }
+
+    private static void UpdateSwitchSize(TMSwitch tMSwitch)
+    {
+        tMSwitch.border.Padding = 6;
+        if (tMSwitch.SwitchSize == SwitchSize.Medium)
+        {
+            tMSwitch.border.HeightRequest = 24;
+            tMSwitch.border.WidthRequest = 48;
+            tMSwitch.circle.HeightRequest = 12;
+            tMSwitch.circle.WidthRequest = 12;
+        }
+        else
+        {
+            tMSwitch.border.HeightRequest = 32;
+            tMSwitch.border.WidthRequest = 64;
+            tMSwitch.circle.HeightRequest = 16;
+            tMSwitch.circle.WidthRequest = 16;
+        }
+    }
+
+    private void OnSwitchTapped(object sender, TappedEventArgs e)
+    {
+        _switchSelected = !_switchSelected;
+        if (!_switchSelected)
+        {
+            OnSwitchUnSelected(this);
+        }
+        else
+        {
+            OnSwitchSelected(this);
+        }
+    }
+    private static void OnSwitchSelected(TMSwitch tMSwitch)
+    {
+        tMSwitch._clicked?.Invoke(tMSwitch, tMSwitch.switchEventTrue);
+        tMSwitch.ToggledCommand?.Execute(tMSwitch.switchEventTrue);
+
+        if (tMSwitch.SwitchSize == SwitchSize.Medium)
+        {
+            tMSwitch.Animate("SelectedAnimation",
+                new Animation
+                {
+            { 0, 1, new Animation(v => tMSwitch.circle.Scale = v, 1.0, 1.36) },
+            { 0, 1, new Animation(v => tMSwitch.circle.TranslationX = v, 0,tMSwitch.border.Width - (tMSwitch.circle.Width +13.6) ) }
+                },
+                length: 250, easing: Easing.CubicIn);
+        }
+        else
+        {
+            tMSwitch.Animate("SelectedAnimation",
+                new Animation
+                {
+            { 0, 1, new Animation(v => tMSwitch.circle.Scale = v, 1.0, 1.38) },
+            { 0, 1, new Animation(v => tMSwitch.circle.TranslationX = v, 0,tMSwitch.border.Width - (tMSwitch.circle.Width +14.8) ) }
+                },
+                length: 250, easing: Easing.CubicIn);
+        }
+        tMSwitch.border.BackgroundColor = ResourcesDictionary.ColorsDictionary(ColorsConstants.SwitchSelected);
+        tMSwitch.border.Padding = 6;
+    }
+    private static void OnSwitchUnSelected(TMSwitch tMSwitch)
+    {
+        tMSwitch._clicked?.Invoke(tMSwitch, tMSwitch.switchEventFalse);
+        tMSwitch.ToggledCommand?.Execute(tMSwitch.switchEventFalse);
+        tMSwitch.Animate("UnselectedAnimation",
+            new Animation
+            {
+            { 0, 1, new Animation(v => tMSwitch.circle.Scale = v, 1.3, 1.0) },
```

Comment:
Here too. Magic numbers

Reply:
Here too logic is changed now so no need to define variables.